### ISSUE TITLE
Editor: Improve template search (ctrl-f)

### ISSUE THREAD

**Issue by BenTangible** [TangibleInc/template-system#80]
The built-in search-replace option sorta works, but there are a few issues with it:

- In many themes, the contrast between the highlighted search result, the text, and the editor background isn't great. This has already been highlighted in issue #55 and I imagine the solution here would be to simply curate one or two light/dark themes with usable text/highlight contrast.
- The search bar is stuck to the bottom of the editor, which means that if you click search/previous/next, the view will scroll up to the current search result and you can no longer access the next button without scrolling back down.
- The WordPress admin bar hides the search result due to there being no scroll offset. See screenshots below where I searched for the term "spartan".
![image](https://github.com/TangibleInc/template-system/assets/113368575/ba060609-e619-4575-b73b-fb224fb3deae)
![image](https://github.com/TangibleInc/template-system/assets/113368575/1a1a7d9f-9cff-437c-8d85-7dfc97e80038)

These wouldn't be such a significant issue if Codemirror wasn't also incompatible with the browser's own built-in search option. The behavior here is really erratic so I don't know exactly what the rules are, but it seems like Codemirror may only be loading the portion of the template inside or above the current viewport. So in the template above, if I'm at the top of the page and don't have my typing cursor within the template editor and hit ctrl-f to find results for the term "spartan",  it shows that there are zero results. However, when I scroll to the bottom and try again, it finds the word and scrolls me up to it.

I imagine the behavior and placement of the built-in Codemirror search would take a fair amount of effort to improve, so I wonder if a short-term solution here would be to make it so that the full editor content is loaded when viewing an L&L template so that at least the browser's own search functionality can be used.

Sorry that I'm discovering so many minor issues these days 😅. Most of them aren't urgent, I'm just documenting them as I find them.

**Comment by eliot-akira** [comment]
I haven't yet explored CodeMirror's search feature much - but I see what you mean, all these issues make it almost unusable. It does sound like it will take effort to get it ready for practical use.

Maybe for the short term, the keyboard shortcut can be changed to let the browser's search function work instead of CodeMirror's. When the latter is in good working condition, it can be switched back.

---

> Most of them aren't urgent, I'm just documenting them as I find them.

Sure, I'm very glad you're putting the whole system to the test and noting down the issues. It's great for quality assurance to have a "power user" who is pushing the limits. ;)

![testing_meme12](https://github.com/TangibleInc/template-system/assets/5352434/7725032d-f4b1-44c8-9144-2fa0cf40373e)

**Comment by BenTangible** [comment]
> Maybe for the short term, the keyboard shortcut can be changed to let the browser's search function work instead of CodeMirror's.

Yeah, that's what I'm thinking. We'd "just" need to make sure the full template is visible to the browser's search function instead of only the portion that's currently in the viewport.
